csv_generator/process_xml_zip.py

```diff
@@ -63,6 +63,45 @@ def get_file_list(soup: BeautifulSoup) -> List[str]:
     return [name.contents[0] for name in soup.find_all('file_nm')]
 
 
+def process_extracted_dir(source_zip, extracted_dir, output_dir, batch=False):
+    go_soup = make_soup(os.path.join(extracted_dir, GO_XML_FILE_NAME))
+
+    create_date = timestamp_to_epoch(get_create_date(go_soup))
+
+    feed_consumers(
+        file_list=tqdm(get_file_list(go_soup), leave=False, disable=batch),
+        output_dir=output_dir,
+        create_date=create_date,
+        zip_dir=extracted_dir,
+        zip_file_name=source_zip.split(os.path.sep)[-1]
+    )
+
+
+def process_zip(source_zip, output_dir, batch=False):
+    extract_zip_file(source_zip=source_zip, output_dir=DEFAULT_ZIP_OUTPUT_DIR)
+
+    process_extracted_dir(
+        source_zip, DEFAULT_ZIP_OUTPUT_DIR, output_dir,
+        batch=batch
+    )
+
+    clean_up_zip(target_dir=DEFAULT_ZIP_OUTPUT_DIR)
+
+def process_zip_or_extracted_dir(source_zip_or_extracted_dir, output_dir, batch=False):
+    if os.path.isdir(source_zip_or_extracted_dir):
+        if not os.path.exists(os.path.join(source_zip_or_extracted_dir, GO_XML_FILE_NAME)):
+            raise RuntimeError('source-zip is a directory but does not contain %s' % (
+                GO_XML_FILE_NAME
+            ))
+        process_extracted_dir(
+            source_zip_or_extracted_dir,
+            source_zip_or_extracted_dir,
+            output_dir,
+            batch=batch
+        )
+    else:
+        process_zip(source_zip_or_extracted_dir, output_dir, batch=batch)
```

Comment:
I think the naming of `source_zip_or_extracted_dir` slightly confuses/hides whats going on in `process_zip_or_extracted_dir`. I can follow it, but maybe renaming it locally after finding out what it is you have been passed (`source_zip` or `extracted_dir`) might make it clearer.

Reply:
Okay, I will assign it to local variables.